### DISCUSSION TITLE
PnP (v4) Search Webpart Not Showing Home.aspx page in Search Results - SharePoint Online

### DISCUSSION THREAD

**Discussion by ChiragDSS** [microsoft-search/pnp-modern-search#4160] [Q&A]
Hi All,

Looking for PnP Search Results Support - SharePoint Online

I have a search page in which I have used PnP V4 Search Webpart to fetch pages from a SitePages Library. SitePages Library has 3 pages created under it, all published, with no unique permissions, visible to all.

I used the below query to show the 3 pages in search results.

{searchTerms} (Path: https://company.sharepoint.com/sites/Functions/Functions_SubSite/SitePages/)

However, Page1 and Page2 are visible, however Page3 with name 'Home.aspx' is not visible ( I used 'Make Homepage' option)

Any suggestions??

**Comment by kasperbolarsen** [comment]
Try to see what the API returns using one of the tools listed here https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/

**Comment by jblueml** [comment]
There seems to be a special treatment when crawling "Homepages".

The “Homepage” of a site does not exist in the search index with managed property value “`contentclass=STS_ListItem_WebPageLibrary`”, but instead with the value “`contentclass=STS_Site`”. 

In other words: the crawled Homepage-item is not being treated as a "List-Item" in the search index, but instead as a "Site".

Furthermore, the managed property “`Path`” in the search index just contains the URL of the site itself and _not_ the absolute path to the appropriate .aspx file (e.g. Home.aspx, TopicHome.aspx, etc.) 
--> in your case: just https://company.sharepoint.com/sites/Functions/Functions_SubSite

And that's the reason, why your query doesn't cover all expected results when you query the path https://.../SitePages/.

But you could give the following query a try:
`{searchTerms} (Path:https://company.sharepoint.com/sites/Functions/Functions_SubSite ContentTypeId:0x0101009D1CB255DA76424F860D91F20E6C4118*)`

_Explanation_:
- omit the /SitePages/ part to just narrow down results to your site
-  `0x0101009D1CB255DA76424F860D91F20E6C4118` is the ID of SharePoint's base site-content-type "Site Page"

**Comment by wobba** [comment]
This is legacy "by design" in SharePoint. The home page of a site will always be indexed as a site entity. The only way to change this if you want the site+page is by using a "hack" I do not recommend which is to break permission inheritance on the home page. If you do this you will get one result on the site result and one on the home page result.

If you want to treat the home page as a page, then include it using `(contentclass:sts_site OR contentclass:sts_web)` in your KQL and render it like you would a page.
